crates/component/src/wrappers/io.rs

```diff
@@ -119,6 +119,26 @@ impl Default for StdioStream<'_> {
     }
 }
 
+/// Similar to [`crate::wasi::io::poll::poll`], but polls all `pollables` until they are all ready.
+///
+/// Poll for completion on a set of pollables.
+///
+/// This function takes a list of pollables, which identify I/O sources of interest, and waits until all of the events are ready for I/O.
+pub fn poll_all(pollables: &[&crate::wasi::io::poll::Pollable]) {
```

Comment:
Will have to test with a `with: {}` block in order to see this working with wasmcloud_component 😄